c/tskit/tables.c

```diff
@@ -8105,6 +8105,285 @@ tsk_table_collection_subset(
     return ret;
 }
 
+static int
+tsk_check_subset_equality(tsk_table_collection_t *self, tsk_table_collection_t *other,
+    tsk_id_t *node_mapping, tsk_size_t other_num_nodes, tsk_size_t num_shared_nodes)
+{
+    int ret = 0;
+    tsk_id_t k, i;
+    tsk_id_t *self_nodes = NULL;
+    tsk_id_t *other_nodes = NULL;
+    tsk_table_collection_t self_copy;
+    tsk_table_collection_t other_copy;
```

Comment:
Need to memset these table collections to zero before any goto. The thing you have to keep in mind is, has everything been initialised **before** the first goto? If not, then we have undefined behaviour. The simplest thing to do here is 
```
memset(&self_copy, 0, sizeof(self_copy));
memset(&self_copy, 0, sizeof(other_copy));
```
at the start.